internal/controller/postgrescluster/cluster.go

```diff
@@ -142,11 +142,13 @@ func (r *Reconciler) generateClusterPrimaryService(
 	service.Spec.ClusterIP = corev1.ClusterIPNone
 	service.Spec.Selector = nil
 
+	appProtocol := naming.AppProtocolTCP
 	service.Spec.Ports = []corev1.ServicePort{{
 		Name:       naming.PortPostgreSQL,
 		Port:       *cluster.Spec.Port,
 		Protocol:   corev1.ProtocolTCP,
 		TargetPort: intstr.FromString(naming.PortPostgreSQL),
+		AppProtocol: &appProtocol,
```

Comment:
⛏️ We have a package that can turn expressions into pointers. I recommend `AppProtocol: initialize.String(…),` here.

https://github.com/CrunchyData/postgres-operator/blob/dd408ff606a3754bf008ef80e40400dbb9a95c51/internal/initialize/primitives.go#L34-L35

Reply:
#3437 merged, so this might be:
```suggestion
		AppProtocol: initialize.String(postgres.IANAServiceName),
```

Reply:
updated